Description/DescriptionFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Resource\Description;
+
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
+
+class DescriptionFactory
+{
+    private $enrichers = [];
+
+    /**
+     * @param array $enrichers
+     */
+    public function __construct(array $enrichers)
+    {
+        $this->enrichers = $enrichers;
+    }
+
+    /**
+     * Return a description of the given (CMF) Resource.
+     *
+     * @param CmfResource $resource
+     */
+    public function getPayloadDescriptionFor(CmfResource $resource)
+    {
+        $type = $resource->getPayloadType();
+        $payload = $resource->getPayload();
```

Comment:
we have to check if payload is an object. At the moment, it can be anything: https://github.com/symfony-cmf/resource/blob/master/Repository/Resource/CmfResource.php#L40-L48 (I even believe we use it as string in the ResourceBundle atm)